src/Transport/Topology/MetaModel/NamespaceInfo.cs

```diff
@@ -1,45 +1,36 @@
 ﻿namespace NServiceBus.AzureServiceBus
 {
-    public class NamespaceInfo
+    using System;
+
+    // NOTE: Can namespaces switch mode?
```

Comment:
Ah ok, got it: we don't need to change `Mode` property of the namespace instance and NamespaceInfo can be a "value object". 
f.e. for `ShardedNamespacePartitioningStrategy` we create a new NamespaceInfo instance (with right values) each time